File: stack&queue/programmers_42586/solution.js
function solution(progresses, speeds) {
    const answer = [];

    const days = progresses.map((progress, index) => 
        Math.ceil((100 - progress) / speeds[index])
    );

    let maxDay = days[0];
    let count = 1;

    for (let i = 1; i < days.length; i++) {
        if (days[i] <= maxDay) {
            count++;
        } else {
            answer.push(count);
            maxDay = days[i];
            count = 1;
        }
    }

    answer.push(count);

    return answer;
}

File: stack&queue/programmers_42586/solution.py
import math

def solution(progresses, speeds):
    answer = []
    
    days = [math.ceil((100 - a) / b) for a, b in zip(progresses, speeds)]

    maxDay = days[0]
    count = 1
    
    for i in range(1, len(days)):
        if days[i] <= maxDay:
            count += 1
        else:
            answer.append(count)
            maxDay = days[i]
            count = 1

    answer.append(count)
            
    return answer